presto-client/src/main/java/com/facebook/presto/client/ClientSession.java

```diff
@@ -103,6 +105,7 @@ public ClientSession(
         this.preparedStatements = ImmutableMap.copyOf(requireNonNull(preparedStatements, "preparedStatements is null"));
         this.roles = ImmutableMap.copyOf(requireNonNull(roles, "roles is null"));
         this.extraCredentials = ImmutableMap.copyOf(requireNonNull(extraCredentials, "extraCredentials is null"));
+        this.customHeaders = ImmutableMap.copyOf(requireNonNull(customHeaders, "customHeaders is null"));
```

Comment:
do we need a customHeader passed in mandatorily.?
Would an Optional suffice instead?

Reply:
We can do the optional, but it seems there is no trend here
What I mean is that extraCredentials can be empty as well, but it is not optional. 
Will it impact the coding style of Presto?

Reply:
lets keep it as Map for now in accordance with the other properties we are setting.